File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/security/Instantiate.java
package net.dontdrinkandroot.gitki.wicket.security;

import net.dontdrinkandroot.gitki.model.Role;

import java.lang.annotation.*;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PACKAGE, ElementType.TYPE})
@Documented
@Inherited
public @interface Instantiate
{
    Role value();

    boolean allowAnonymousIfConfigured() default false;
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/security/Instantiate.kt
package net.dontdrinkandroot.gitki.wicket.security

import net.dontdrinkandroot.gitki.model.Role
import java.lang.annotation.Inherited

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FILE, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@MustBeDocumented
@Inherited
annotation class Instantiate(val value: Role, val allowAnonymousIfConfigured: Boolean = false)